esp-hal/src/clock/mod.rs

```diff
@@ -67,7 +67,7 @@
 //! // });
 //! //
 //! // Initialize with default clock frequency for this chip
-//! // let peripherals = esp_hal::init(esp_hal::Config::default());
+//! // let peripherals = esp_hal::init(esp_hal::config::Config::default());
```

Comment:
imo we should reexport the global `Config` in lib.rs to avoid this extra namespace.

Reply:
We have like half a dozen `Config` structs, I'm kind of leaning towards keeping the namespace for this reason.

Reply:
Ups, just reverted before reading @jessebraham comment, happy to revert the revert commit haha. Don't have a hard opinion on this matter.

Reply:
Maybe `config::InitConfig`?

Reply:
please don't make this more painful than `esp_hal::Config`